docs: add products.ts file on getting started page

ajitsinghkaler opened the pull request:
Fixes #34291

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
In the current behavior, products.ts page is not visible on the getting started page.

Issue Number: 34291


## What is the new behavior?
Products.ts page is visible on the new getting started page.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

ajitsinghkaler commented:
@gkalpak can ou please review this.

mary-poppins commented:
You can preview 7c02a49 at https://pr34301-7c02a49.ngbuilds.io/.
You can preview efe649e at https://pr34301-efe649e.ngbuilds.io/.

kapunahelewong commented on aio/content/start/index.md:
```diff
@@ -80,8 +80,14 @@ To help you get going, the following steps use predefined product data and metho
 
     1. Each product in the list displays the same way, one after another on the page. To iterate over the predefined list of products, put the `*ngFor` directive on a `<div>`, as follows:
```
Since we're adding a file, but the reader doesn't need to copy the contents of `products.ts` into their project, we should point it out. Maybe something like the following would be helpful:

```suggestion
    1. When you generated the app, StackBlitz automatically created a file called `products.ts`, with a list of products to use in your app. Each product in the list displays the same way, one after another on the page. To iterate over the predefined list of products, put the `*ngFor` directive on a `<div>`, as follows:
```

ajitsinghkaler replied:
I've made the changes please have a look

kapunahelewong reviewed (changes requested):
Thank you for this, @ajitsinghkaler. Just one comment.

mary-poppins commented:
You can preview 6a2ff77 at https://pr34301-6a2ff77.ngbuilds.io/.

gkalpak commented on aio/content/start/index.md:
```diff
@@ -78,10 +78,16 @@ To help you get going, the following steps use predefined product data and metho
+        <code-pane header="src/app/product-list/product-list.component.html" path="getting-started/src/app/product-list/product-list.component.2.html"  region="ngfor">
+        </code-pane>
+
+        <code-pane header="src/app/products.ts" path="getting-started/src/app/products.ts"></code-pane>
```
FWIW, I don't think this change addresses #34291. #34291 refers to the sentence:

> The product list component also defines a `products` property that contains imported data for each product from the `products` array in `products.ts`.

This sentence appears in the ["Input" section][1], further below, so adding the `products.ts` file here (before it is being mentioned) would be more confusing imo.

##
Personally, I don't agree with the suggestion in #34291 that `products.ts` should be shown in the tutorial just because it is part of the tutorial. There are many files that are part of it, but not shown in the tutorial. All files are available on the StackBlitz project, so the user can look at them if they want to.

In any case, if we are going to include `products.ts` in a code-pane, it should either be somewhere around the [Input section][1] (where it is mentioned) or we should mention it earlier.

[1]: https://angular.io/start#input

kapunahelewong replied:
Agreed with @gkalpak. I don't think it's necessary to add it, but if we do, it should definitely be mentioned (wherever it may go). I see why @ajitsinghkaler added it here, though, since those are the products we're iterating through. 

@ajitsinghkaler what are your thoughts about your change? Do you feel it enhances clarity?

ajitsinghkaler replied:
@kapunahelewong I get what @gkalpak is trying to tell even I was not sure to sow the product.ts file there but my concern is this is the first tutorial we should mostly spoon feed it in there if it was one of the later tutorials it was okay to not show it but I'm not sure about the starter. Or what we can do is add a line in the stack blitz Tips where we explain the starter files. That seems like a more elegant solution

@gkalpak I thought #34291 would be resolved if we show the file once at the top it won't be a problem anymore, showing it the first time it was mentioned should be enough.

I need your thoughts on this.

kapunahelewong replied:
Adding a line about it in the StackBlitz notes section could be helpful.

ajitsinghkaler replied:
I'll add another line regarding products.ts in the stackblitz notes

ajitsinghkaler commented:
@kapunahelewong I have made some changes can you please have a look.

mary-poppins commented:
You can preview 628eb3c at https://pr34301-628eb3c.ngbuilds.io/.

kapunahelewong commented on aio/content/start/index.md:
```diff
@@ -50,6 +50,9 @@ expect, save and then click the refresh button.
+* The tutorial may contain some file references straight
+from the StackBlitz examples. If you can't find some files
+please have a look in the examples.
 
```
How about:

> When you generate the StackBlitz example app that accompanies the Getting Started, StackBlitz creates the starter files and mock data for you. The files you'll use throughout the Getting Started are in the `src` folder of the StackBlitz example app.

kapunahelewong replied:
So, similar message but the emphasis is on where to find the files rather than not being able to find them. Does that help?

ajitsinghkaler replied:
@kapunahelewong I think we can put this as a general message rather than it being just about Getting started example.

ajitsinghkaler replied:
Because there we are giving tips about how to use stackblitz for all example apps I've added other text for the products.ts file.

kapunahelewong replied:
Makes sense to me. @ajitsinghkaler, could you put `src` in backticks so it shows up in code font?

ajitsinghkaler replied:
@kapunahelewong Can you please have a look I put in the back ticks

mary-poppins commented:
You can preview 342817b at https://pr34301-342817b.ngbuilds.io/.

ajitsinghkaler committed:
docs: add products.ts file on getting started page

in the getting started page (first tutorial) file products.ts which was not shown and was only present in the StackBlitz examples. So added a refrence that it is present in the example and also added a note that examples may carry filenames not present please look at StackBliz examples for details

Fixes #34291

mary-poppins commented:
You can preview af02f3d at https://pr34301-af02f3d.ngbuilds.io/.

kapunahelewong reviewed (approved):
LGTM. Thanks, @ajitsinghkaler.

Let's see what @gkalpak has as feedback as our next step.

gkalpak reviewed (approved):
LGTM2 :+1:

ajitsinghkaler commented:
@kapunahelewong now can we add the appropriate labels to this

ajitsinghkaler commented:
@gkalpak can you please merge this

gkalpak commented:
@ajitsinghkaler: I didn't realize it didn't have the merge label. Added. Thx for the ping.
(BTW, please do not re-request reviews, unless you really need the person to review the code again. It is confusing.)

gkalpak commented:
**merge-assistance:** Global approval for this docs-only change.
